lib/internal/http2/compat.js

```diff
@@ -223,6 +222,27 @@ const proxySocketHandler = {
   }
 };
 
+function onStreamCloseRequest() {
+  const req = this[kRequest];
+  const state = req[kState];
+
+  if (!this[kRequest])
```

Comment:
`req === undefined` is more precise. Also please move `const state =` after the check.